nixos/modules/services/x11/display-managers/sddm.nix

```diff
@@ -14,7 +14,7 @@ let
   xserverWrapper = pkgs.writeScript "xserver-wrapper" ''
     #!/bin/sh
     ${concatMapStrings (n: "export ${n}=\"${getAttr n xEnv}\"\n") (attrNames xEnv)}
-    exec ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
+    exec systemd-cat ${dmcfg.xserverBin} ${dmcfg.xserverArgs} "$@"
```

Comment:
No this is almost certainly worse than the current situation. Have a look at the initial issue that lead to removal of `systemd-cat` in that line:
https://github.com/NixOS/nixpkgs/issues/16096

@abbradar clearly states that applications __fail__ when the pipe is lost.
So we have either
- no logging, or
- applications crashing after system rebuild

i vote for the first option.